netsim/modules/_routing.py

```diff
@@ -321,7 +321,9 @@ def build_vrf_interface_list(
     for neighbor in l.neighbors:                                      # ... iterate over the list of neighbors
       n_data = topology.nodes[neighbor.node]
       if proto in n_data.get('module',[]):                            # ... and check if at least one of them uses the IGP
-        node.vrfs[l.vrf][proto].active = True
+        for af in ['ipv4','ipv6']:                                    # ... and has at least one AF in common
+          if af in l and af in neighbor:                              # ... with our interface
+            node.vrfs[l.vrf][proto].active = True                     # Found one? Let's keep the routing protocol active
```

Comment:
Should we track this by address family? OSPFv2 and OSPFv3 are not the same thing...

Reply:
Can break the outer for loop once the flag has been set

Reply:
> Should we track this by address family? OSPFv2 and OSPFv3 are not the same thing...

The address families are tracked elsewhere (based on what is configured on global/VRF interfaces). Here, we're just checking whether there's an AF match between current device and its neighbors.

However, the "for" loop could be rewritten as "for af in ...[proto].af" if the **af** dictionary is already set at that moment. Thanks for the suggestion: have to check...